tests/bugfixes/github/test_canon_lenses.py

```diff
@@ -0,0 +1,390 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+import system_tests
+
+
+# List of all supported lenses, duplicated from canonCsLensType in src/canonmn_int.cpp,
+# to ensure that lenses do not get removed accidentially.
+# Lenses should never be removed from this list to guarantee backwards-compatibility.
+lenses = '''
```

Comment:
@webmeister instead of having two lists of these lenses which have to be kept in sync manually, what do you think about just using python to query this list automatically from `src/canonmn_int.cpp`

Reply:
I thought about this too, but then decided against it. The purpose of this test is to prevent regressions in the lens detection code, i.e. all lenses it detects today shall be detected correctly in the future. If it read the list of lenses from `src/canonmn_int.cpp`, you could remove half the lenses currently listed there and the test would still pass. Therefore, I chose to duplicate the list here and added the comment above to explain its purpose. This way, at least it needs to be a deliberate decision to remove support for a lens (which might happen if a manufacturer decides to produce a lens with the exact same metadata as an existing lens, so that it is impossible to tell them apart).

Reply:
Ok, fair enough, I see your point ;) Let's leave the list in.  

How about then simply adding a comparison of the two lists to print a warning if they diverge?  
My main motivation here is that I think this is a great test you introduced here, and I'd like to avoid anyone adding a new lens and forgetting to add it to this test, as, IMHO, part of its strength is that it is supposed to test all lenses.